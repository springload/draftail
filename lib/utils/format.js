// From http://stackoverflow.com/a/10903003/1798491.
const shortString = (s, l, reverse = false) => {
    const stopChars = [' ', '/', '&'];
    const acceptableShortness = l * 0.8; // When to start looking for stop characters
    const str = reverse ? s.split('').reverse().join('') : s;
    let ret = '';

    let i;
    for (i = 0; i < l - 1; i++) {
        ret += str[i];
        if (i >= acceptableShortness && stopChars.indexOf(str[i]) >= 0) {
            break;
        }
    }

    if (reverse) {
        return ret.split('').reverse().join('');
    }
    return ret;
};

// From http://stackoverflow.com/a/10903003/1798491.
export const truncateURL = (url, l = 50) => {
    const chunkLength = l / 2;
    let ret = url.replace('http://', '').replace('https://', '');

    if (url.length > l) {
        const start = shortString(ret, chunkLength, false);
        const end = shortString(ret, chunkLength, true);

        ret = `${start}â€¦${end}`;
    }

    return ret;
};
