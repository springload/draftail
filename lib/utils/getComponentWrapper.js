import React from "react";

/**
 * Wraps a component to provide it with additional props based on context.
 */
const getComponentWrapper = (Wrapped, wrapperProps) => {
  const Wrapper = (props) => <Wrapped {...props} {...wrapperProps} />;

  return Wrapper;
};

export default getComponentWrapper;
