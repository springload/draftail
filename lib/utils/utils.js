import React from 'react';

/**
 * Wraps a component to provide it with additional props based on context.
 */
export const getComponentWrapper = (Wrapped, wrapperProps) => {
    const Wrapper = props => <Wrapped {...props} {...wrapperProps} />;

    return Wrapper;
};
