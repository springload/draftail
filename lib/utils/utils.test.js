import React from 'react';
import { shallow } from 'enzyme';

import { getComponentWrapper } from '../utils/utils';

describe('utils', () => {
    it('#getComponentWrapper', () => {
        const Wrapped = () => <div />;
        const Wrapper = getComponentWrapper(Wrapped, {
            test: true,
        });

        expect(shallow(<Wrapper />)).toMatchSnapshot();
    });
});
