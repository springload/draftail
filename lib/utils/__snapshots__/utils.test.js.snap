// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils #getComponentWrapper 1`] = `
<Wrapped
  test={true}
/>
`;
